substrate/client/network/sync/src/chain_sync.rs

```diff
@@ -240,14 +243,6 @@ pub struct ChainSync<B: BlockT, Client> {
 	downloaded_blocks: usize,
 	/// State sync in progress, if any.
 	state_sync: Option<StateSync<B, Client>>,
```

Comment:
Because of the "fast" aka "light state" sync. `ChainSync` is responsible for it, and after downloading block history (without executing blocks) it does state sync and then continues with normal block sync (executing blocks). It should be possible to refactor it in a way that we switch the startegies in a following way:
1. `ChainSync` without executing blocks
2. `StateSync`
3. "Normal" `ChainSync`

But I'm not sure if it makes sense, considering it's mostly `ChainSync` with a small period of `StateSync`.
We can revisit this in a follow-up though.

Reply:
Yes I think we have to think about this a little so a follow-up is a good idea